packaging/flavorgen/cloudprovider/cloud-controller-manager.go

```diff
@@ -35,295 +35,63 @@ const (
-func CloudControllerManagerServiceAccount() *corev1.ServiceAccount {
-	return &corev1.ServiceAccount{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "cloud-controller-manager",
-			Namespace: "kube-system",
-		},
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "ServiceAccount",
-			APIVersion: corev1.SchemeGroupVersion.String(),
-		},
+// CloudControllerManagerManifests returns the yaml manifests generated via
+// `helm template` from https://github.com/kubernetes/cloud-provider-vsphere/tree/master/charts/vsphere-cpi
+func CloudControllerManagerManifests() (string, error) {
+	objectList := []string{}
+
+	imageTagRemoval := regexp.MustCompile(`(image: .*):.*`)
+	withoutTag := imageTagRemoval.ReplaceAll([]byte(cpiManifests), []byte(`${1}:$`+env.CPIImageKubernetesVersionVar))
+
+	// ClusterResourceSet does not support `kind: List`, we have to adjust the manifests
+	// to remove those lists and directly add the items instead.
+	for _, object := range strings.Split(string(withoutTag), "---\n") {
+		// Ignore empty strings
+		if object == "" {
+			continue
+		}
+		// Marshal to unstructured so we can check the type
+		u := &unstructured.Unstructured{}
+		if err := yaml.Unmarshal([]byte(object), u); err != nil {
+			return "", err
+		}
+		// If it is not a List we can directly add the object to the slice
+		if u.GetKind() != "List" {
+			objectList = append(objectList, object)
+			continue
+		}
+
+		// If it is a list we extract the slice at `items`
+		items, isSlice, err := unstructured.NestedSlice(u.UnstructuredContent(), "items")
+		if err != nil {
+			return "", err
+		}
+		if !isSlice {
+			return "", errors.New("expected List.items to be a slice")
+		}
+
+		// Loop over all items, marshal them and add it to objects.
+		for _, item := range items {
+			marshaledItem, err := yaml.Marshal(item)
+			if err != nil {
+				return "", err
+			}
+			objectList = append(objectList, string(marshaledItem))
+		}
 	}
-}
 
-// CloudControllerManagerService returns a Service for the cloud-controller-manager.
-func CloudControllerManagerService() *corev1.Service {
-	return &corev1.Service{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "cloud-controller-manager",
-			Namespace: "kube-system",
-			Labels: map[string]string{
-				"component": "cloud-controller-manager",
-			},
-		},
-		Spec: corev1.ServiceSpec{
-			Type: corev1.ServiceTypeNodePort,
-			Ports: []corev1.ServicePort{
-				{
-					Port:       443,
-					Protocol:   corev1.ProtocolTCP,
-					TargetPort: intstr.FromInt(43001),
-				},
-			},
-			Selector: map[string]string{
-				"component": "cloud-controller-manager",
-			},
-		},
-	}
+	return "---\n" + strings.Join(objectList, "---\n"), nil
 }
 
 // CloudControllerManagerConfigMap returns a ConfigMap containing data for the cloud config file.
 func CloudControllerManagerConfigMap(cloudConfig string) *corev1.ConfigMap {
 	return &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "vsphere-cloud-config",
+			Name:      "cloud-config",
```

Comment:
I assume this means the name in our structs was out-of-sync with the chart, right?

Reply:
Yes, or manually edited so it has the prefix.